rosbag2_storage/include/rosbag2_storage/yaml.hpp

```diff
@@ -41,6 +42,27 @@ void optional_assign(const Node & node, std::string field, T & assign_to)
   }
 }
 
+template<>
+struct convert<std::unordered_map<std::string, std::string>>
+{
+  static Node encode(const std::unordered_map<std::string, std::string> & custom_data)
+  {
+    Node node;
+    for (const auto & it : custom_data) {
+      node[it.first] = it.second;
+    }
+    return node;
+  }
+
+  static bool decode(const Node & node, std::unordered_map<std::string, std::string> & custom_data)
+  {
+    for (YAML::const_iterator it = node.begin(); it != node.end(); ++it) {
+      custom_data.emplace(it->first.as<std::string>(), it->second.as<std::string>());
```

Comment:
`emplace` can fail to create new element. Need to add a sanity check for `emplace` result 
e.g.
```cpp
      if (!custom_data.emplace(it->first.as<std::string>(), it->second.as<std::string>()).second)
      {
        return false;
      }
```

Reply:
Since this method simply decode a yamlcpp Node to a `std::map`, I'm wondering in what case the `emplace` will fail? Note that the internal data structure in yamlcpp Node can already guarantee the uniqueness.

Reply:
From https://en.cppreference.com/w/cpp/container/unordered_map/emplace
```cpp
template< class... Args >
std::pair<iterator,bool> emplace( Args&&... args );
```
> Return value
> Returns a pair consisting of an iterator to the inserted element, or the already-existing element if no insertion happened, and a bool denoting whether the insertion took place (true if insertion happened, false if it did not).

Reply:
Yes that is indeed the return value of `emplace` method. However, like I said, since we are decoding a YAML-cpp node to a STL map, the case of duplicate key, i.e., insertion failure, won't happen because YAML-cpp already did the deduplication.